src/scmdata/groupby.py

```diff
@@ -114,9 +123,67 @@ def apply(self, func, *args, **kwargs):
             The result of splitting, applying and combining this array.
         """
         grouped = self._iter_grouped()
-        applied = [
-            _maybe_wrap_array(arr, func(arr, *args, **kwargs)) for arr in grouped
-        ]
+        applied = [func(arr, *args, **kwargs) for arr in grouped]
+        return self._combine(applied)
+
+    def apply_parallel(
+        self,
+        func: "Callable[Concatenate[GenericRun, P], Union[GenericRun, pd.DataFrame, None]]",
+        n_jobs: int = 1,
+        backend: str = "loky",
+        *args: "P.args",
+        **kwargs: "P.kwargs",
+    ) -> "GenericRun":
+        """
+        Apply a function to each group in parallel and append the results
+
+        Provides the same functionality as :func:`~apply` except that :mod:`joblib` is used to apply
+        `func` to each group in parallel. This can be slower than using :func:`~apply` for small
+        numbers of groups or in the case where `func` is fast as there is overhead setting up the
+        processing pool.
+
+        See Also
+        --------
+        :func:`~apply`
+
+        Parameters
+        ----------
+        func
+            Callable to apply to each timeseries.
+
+        n_jobs
+            Number of jobs to run in parallel (defaults to a single job which is useful for
+            debugging purposes). If `-1` all CPUs are used.
```

Comment:
A flip of a coin. My thoughts were that you explicitly opt into how many resources you use, but you kind of already signal that you want to parallelise by opting to use `apply_parallel` instead of `apply`. It is also the default for joblib.Parallel